pkg/services/ngalert/state/compat.go

```diff
@@ -71,11 +72,19 @@ func StateToPostableAlert(alertState *State, appURL *url.URL) *models.PostableAl
 		urlStr = ""
 	}
 
-	if alertState.State == eval.NoData {
+	state := alertState.State
+	if alertState.Resolved && state == eval.Normal {
```

Comment:
```suggestion
	if alertState.Resolved {
```

What would happen if we just checked `alertState.Resolved`? It should be an invariant that this cannot be true while state is something other than `eval.Normal`, right?

Reply:
I agree with George, this change https://github.com/grafana/grafana/pull/80184/files#diff-6420fa5148996efbf754b18838902601c17b758ab7da6279dc019b5b143fd5c2R220  guarantees that the states will have Resolved=true, and this additional check does not seem to be needed.

Reply:
Fixed: https://github.com/grafana/grafana/pull/80184/commits/e52412da8d2c5cc130f7f7acb49fd810650cfb9b